File: lab04-zeromq/python-hello-world/cliente_hello.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import sys
import zmq
context = zmq.Context()
HOST = sys.argv[1] if len(sys.argv) > 1 else "localhost"
PORT = sys.argv[2] if len(sys.argv) > 2 else "5555"
servidor = "tcp://"+ HOST +":"+ PORT
# Criando o socket para conectar no servidor
s = context.socket(zmq.REQ)
# Fica bloqueado até conseguir se conectar no servidor
s.connect(servidor)
# Envia a mensagem (sequência de bytes)
s.send(b"Hello")
# Fica bloqueado até receber a resposta do servidor
message = s.recv()
# Imprime a resposta do servidor
print("Recebido {}".format(message))

File: lab04-zeromq/java-hello-world/app/src/main/java/engtelecom/std/ClienteHello.java
package engtelecom.std;

import org.zeromq.SocketType;
import org.zeromq.ZContext;
import org.zeromq.ZMQ;

public class ClienteHello {
    public static void main(String[] args) {
        try (ZContext context = new ZContext()) {
            // Socket para conversar com o servidor
            ZMQ.Socket socket = context.createSocket(SocketType.REQ);
            System.out.println("Conectando no servidor Hello World!");
            // Fica bloqueado até conseguir se conectar no servidor
            socket.connect("tcp://localhost:5555");
            System.out.println("Enviando mensagem ");
            String request = "Hello";
            // Envia a mensagem (sequência de bytes)
            socket.send(request.getBytes(ZMQ.CHARSET), 0);
            // Fica bloqueado até receber a resposta do servidor
            byte[] reply = socket.recv(0);
            System.out.println("Recebido " + new String(reply, ZMQ.CHARSET));
        }
    }
}